### DISCUSSION TITLE
Scaling not working as expected

### DISCUSSION THREAD

**Discussion by LudoFarkas** [coil-kt/coil#921] [Q&A]
**Describe the bug**
Hi,

I'm migrating from Glide to Coil and have an issue with image scaling. I'm used to set the scale type in code. So in order to achieve `ImageView.ScaleType = CENTER_CROP` I set `scale(Scale.FILL)`. The problem is that the image does not fill the `ImageView`, but fits entirely in. However, if the scale type is set in xml, it works as expected.

**Expected behavior**
I would like to get a clear explanation of how Coil's scale should work in case I don't understand it right. Or to have it corrected in case there is a bug.

**To Reproduce**
I have 2 `ImageView`s in a layout with blue background to highlight their borders (one of them without explicitly set scale type, the second set to `centerCrop`). The expected behavior is to achieve filling both of the `ImageView`s:

```
<androidx.appcompat.widget.LinearLayoutCompat
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    android:gravity="center"
    android:orientation="vertical">

    <androidx.appcompat.widget.AppCompatImageView
        android:id="@+id/thumbnail1ImageView"
        android:layout_width="200dp"
        android:layout_height="200dp"
        android:background="#19dbff" />

    <androidx.appcompat.widget.AppCompatImageView
        android:id="@+id/thumbnail2ImageView"
        android:layout_width="200dp"
        android:layout_height="200dp"
        android:layout_marginTop="20dp"
        android:background="#19dbff"
        android:scaleType="centerCrop" />
</androidx.appcompat.widget.LinearLayoutCompat>
```

The same image is loaded to them in code:

```
thumbnail1ImageView.load(File(photoPath)){
            scale(Scale.FILL)
}

thumbnail2ImageView.load(File(photoPath))
```

Only the second image fills the `ImageView`, see screenshots. 

**Logs/Screenshots**
![minions](https://user-images.githubusercontent.com/45174293/135852365-a6dc8635-f47f-4dc6-9f64-666d961c7147.jpg)

**Version**
1.3.2

**Comment by colinrtwhite** [comment]
You should set the scale type on the `ImageView`. Coil will auto-detect the scale from the `ImageView`'s scale type.

**Reply by LudoFarkas** [reply]
OK, but then for what use case should be `scale(scale: Scale)` used for?

**Comment by mahmoudelfouly** [comment]
I'm facing the same issue here. Have you got any answer?
